### PR TITLE
Add `DROP ANALZER` statement support.

### PR DESCRIPTION
**Pull request by seut**
## Summary of the changes / Why this improves CrateDB

Implement the `DROP ANALYZER` statement to support removal of custom analyzer definitions from the cluster.

## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: sql/src/main/java/io/crate/analyze/CreateAnalyzerAnalyzedStatement.java [hunk 1]
```diff
@@ -87,59 +89,63 @@ public void extendedAnalyzer(String name) {
...
 
-    public boolean extendsBuiltInAnalyzer() {
-        return extendedAnalyzerName != null && (extendedCustomAnalyzer == null ||
-                                                !extendedCustomAnalyzer.get(String.format(Locale.ENGLISH, "index.analysis.analyzer.%s.type",
-                                                    extendedAnalyzerName)).equals("custom"));
+    private boolean extendsBuiltInAnalyzer() {
+        return extendedAnalyzerName != null
+               && (
+                   extendedCustomAnalyzer == null
+                   ||
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/analyze/CreateAnalyzerAnalyzedStatement.java]: formatting seems a bit odd?

File: sql/src/test/java/io/crate/analyze/DropAnalyzerTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                    .build().toString())
+            .build();
+
+        ClusterState clusterState = ClusterState.builder(clusterService.state())
+            .metaData(MetaData.builder(clusterService.state().metaData())
+                .persistentSettings(settings))
+            .build();
+        ClusterServiceUtils.setState(clusterService, clusterState);
+
+        e = SQLExecutor.builder(clusterService).enableDefaultTables().build();
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/analyze/DropAnalyzerTest.java]: `enableDefaultTables` is probably not required?

File: blackbox/docs/appendices/release-notes/unreleased.rst [hunk 3]
```diff
@@ -42,6 +42,9 @@ None
 Changes
 =======
 
+- Implemented the ``DROP ANALYZER`` statement to support removal of custom
+  analyzer definitions from the cluster.
```
Comment (mfussenegger) [inline blackbox/docs/appendices/release-notes/unreleased.rst]: Should we also mention the `information_schema.routines.routine_definition` change?

File: sql/src/main/java/io/crate/analyze/CreateAnalyzerAnalyzedStatement.java (hunk 1 above)
Comment (seut) [reply sql/src/main/java/io/crate/analyze/CreateAnalyzerAnalyzedStatement.java]: hm personally I like it much more than before. Maybe personal preference? (intellij likes my formatting as well ;-)

File: blackbox/docs/appendices/release-notes/unreleased.rst (hunk 3 above)
Comment (seut) [reply blackbox/docs/appendices/release-notes/unreleased.rst]: Wasn't sure if my solution is fine for everyone. If so, I'd make it visible and add a changes entry.

**Comment by seut** [comment]
@mfussenegger pushed fixups

**Pull request merged**